_ext/css_minifier.rb

```diff
@@ -58,26 +58,9 @@ def transform(site, page, input)
           return input
         end
 
-        oldFileName = File.basename(page.output_path).to_s
-
-        # Create new file name with suffix added
-        newFileName = oldFileName.slice(0..oldFileName.length-4)+"min.css"
-        newOutputPath = File.join(File.dirname(page.output_path.to_s),newFileName)
-
-        # Create a temporary file with the merged content.
-        tmpOutputPath = File.join( "./_tmp/" , newFileName)
-        tmpOutputFile = File.new(tmpOutputPath,"w")
-        tmpOutputFile.write(output)
-        tmpOutputFile.close
-
-        # Add the temporary file to the list of pages for rendering phase.
-        newPage = site.engine.load_page(tmpOutputPath)
-        newPage.source_path = tmpOutputPath
-        newPage.output_path = newOutputPath
-        site.pages << newPage
-
+        page.output_path = page.output_path.gsub('.css', '.min.css')
         # We return the input because we leave the original file untouched
-        input
+        output
```

Comment:
True, but that can be done.

On Wednesday, April 29, 2015, Max Rydahl Andersen <notifications@github.com>
wrote:

> In _ext/css_minifier.rb
> <https://github.com/maxandersen/jbosstools-website/pull/5#discussion_r29322385>
> :
>
> >          # We return the input because we leave the original file untouched
> > -        input
> > +        output
>
> the problem is that this then requires changes to the templates that has
> logic to say "if minified use xyz.min.css if not use xyz.css"
>
> I prefer minifiying in place but this change will need to communicated
> well out to those using something like this to know they need to fix their
> templates too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maxandersen/jbosstools-website/pull/5/files#r29322385>
> .
>


-- 
Jason Porter
http://en.gravatar.com/lightguardjp